qemu/tests/netkvm_protocol_binding.py

```diff
@@ -58,9 +66,20 @@ def run(test, params, env):
 
     test.log.info("Will install inf file found at '%s'", inf_path)
     install_cmd = params["install_cmd"] % inf_path
-    status, output = session.cmd_status_output(install_cmd, timeout=timeout)
-    if status:
-        test.error("Install inf file failed, output=%s" % output)
+
+    # workaround for driver has no signature
+    key_to_install_driver = params.get("key_to_install_driver").split(";")
+    try:
+        session.cmd_status_output(install_cmd, timeout=30)
+    except ShellTimeoutError:
+        send_key(vm, key_to_install_driver)
+        # wait for the installation to complete
+        time.sleep(20)
+    output = session.cmd_output(params["check_installation_cmd"], timeout)
+    if params["check_info"] in output:
+        test.log.info("inf file installed successfully")
+    else:
+        test.error("inf file installed failed")
```

Comment:
```suggestion
    check_session = vm.wait_for_login()
    if not utils_misc.wait_for(
                 lambda: params["check_info"] in check_session.cmd_output(params["check_installation_cmd"]), 30):
        test.error("inf file installed failed")
```

Reply:
Okay, It looks more reasonable, updated the code. Thanks a lot!